generator/src/main/java/com/scottlogic/deg/generator/reducer/ConstraintReducer.java

```diff
@@ -67,23 +64,27 @@ public ConstraintReducer(
     }
 
     public Optional<FieldSpec> reduceConstraintsToFieldSpec(Iterable<AtomicConstraint> rootConstraints,
-                                                            Iterable<AtomicConstraint> decisionConstraints) {
-        if (rootConstraints == null || !rootConstraints.iterator().hasNext()) {
-            return Optional.of(FieldSpec.Empty);
-        }
+                                                            Set<FieldSpec> decisionConstraints) {
```

Comment:
Whats the purpose of this change, is it always a Set that is passed in?

Reply:
the purpose was to change it from `AtomicConstraint` to `FieldSpec` and sets are always passed in